internals/overlord/overlord.go

```diff
@@ -49,6 +49,25 @@ var (
 	defaultCachedDownloads = 5
 )
 
+// Extension represents an extension of the Overlord.
+type Extension interface {
+	// ExtraManagers is called when building an Overlord, it allows additional
+	// managers to be added to the Overlord.
```

Comment:
```
// ExtraManagers allows additional StateManagers to be used.
```

Whether this is called when building the overlord or elsewhere is up to the implementation of the overlord.